repository/src/main/java/com/epam/esm/constructor/QueryConstructor.java

```diff
@@ -0,0 +1,100 @@
+package com.epam.esm.constructor;
+
+import com.epam.esm.constant.SqlGiftCertificateQuery;
+import org.springframework.stereotype.Component;
+
+@Component
+public class QueryConstructor {
+
+    private static final String SELECT_CERTIFICATES_WHERE_MATCH_TAG_NAME_FiRST_PART = "t.name IN ('";
+    private static final String SELECT_CERTIFICATES_WHERE_MATCH_TAG_NAME_SECOND_PART = "') ";
+    private static final String SELECT_CERTIFICATES_WHERE_MATCH_NAME_FIRST_PART = "gc.name like '%";
+    private static final String SELECT_CERTIFICATES_WHERE_MATCH_NAME_SECOND_PART = "%' ";
+    private static final String WHERE = "WHERE ";
+    private static final String AND = "AND ";
+    private static final String ORDER_BY = "ORDER BY ";
+    private static final String COMMA = ", ";
+    private static final String SELECT_CERTIFICATES_WHERE_MATCH_DESCRIPTION_FIRST_PART = "gc.description IN ('";
+    private static final String SELECT_CERTIFICATES_WHERE_MATCH_DESCRIPTION_SECOND_PART = "') ";
+    private static final String SELECT_CERTIFICATES_BY_NAME = "gc.name ";
+    private static final String SELECT_CERTIFICATES_BY_NAME_DESC = "gc.name DESC ";
+    private static final String SELECT_CERTIFICATES_BY_DATE = "gc.last_update_date ";
+    private static final String SELECT_CERTIFICATES_BY_DATE_DESC = "gc.last_update_date DESC ";
+
+
+    public String constructQuery(String tagName, String giftCertificateName, String description, String sortByName,
+                                 String sortByDate) {
+        StringBuilder query = new StringBuilder(SqlGiftCertificateQuery.FIND_ALL_CERTIFICATES);
+
+
+        if (isNotNullThreeParameters(tagName, giftCertificateName, description)) {
```

Comment:
Maybe it will be better to rename method to something like "isAllParametersAreNotNull"? Without adding number of parameters to name.

Reply:
Rename to isAnyParameterNotNull